### DISCUSSION TITLE
compare models

### DISCUSSION THREAD

**Discussion by hl2500** [pycaret/pycaret#3630] [Q&A] [answered]
Hello, I'm new to pycaret. Could I ask 2 questions?

1. Is the final best model, which can be used for making predictions on test and unseen data, from compare_models(), trained on all training data or just 9 folds of the training data from cross validation?
2. Does compare_models() tune the hyperparameters as well? Do we need to implement the tune_model() function?

Thanks!

**Comment by santoshenrique2021** [comment]
I do not see catboost in pycaret

**Reply by ngupta23** [reply]
It is an optional dependency. It does not get installed automatically.

**Reply by hl2500** [reply]
@ngupta23 @santoshenrique2021 Sorry, I didn't understand. I was just wondering if I used 10 fold corss validation, is the output best model from compare_models() trained on all training data or 9 folds of it? Or is it an ensemble of 10 models?

**Reply by ngupta23** [reply, marked as answer]
For scoring, it create 10 models (1 for each fold). This improves the odds of making sure that the metrics are reproducible when we get new data. Then the model that is returned (single model) is trained on all the training data (all folds combined). Hope that helps.

**Reply by hl2500** [reply]
@ngupta23 Thank you for the information! Could I ask if the `fold_strategy=‘kfold’` option in `setup()` function is the same implementation as `sklearn.model_selection.KFold` (based on the given sequence of the data)? For example, if there are 100 data points for a 10 fold cross validation, in the first split, the first 10 data points would be validation and the following 90 data points would be train data?

**Reply by ngupta23** [reply]
Yes, It uses sklearn under the hood.

**Comment by santoshenrique2021** [comment]
Considering classification, I would like to know how to add catboost in pycaret. Furthermore, could you tell me how to compute KS statistics in pycaret? 

Best regards
